File: app/src/main/java/com/mnj/java_solutions/LengthOfLastWordLC58.java
package com.mnj.java_solutions;

public class LengthOfLastWordLC58 {

    public void lengthOfLastWord() {
        String str = "    fly me   to   the moon  ";

        int count = 0;
        for (int i = str.length() - 1; i >= 0; i--) {
            char c = str.charAt(i);
            if (c != ' ') {
                count++;
            } else {
                if (count > 0) {
                    System.out.println("==>> Length of last word :" + count);
                    return;
                }
            }
        }
    }
}

File: app/src/main/java/com/mnj/java_solutions/LengthOfLastWordLC58.kt
package com.mnj.java_solutions

class LengthOfLastWordLC58 {
    fun lengthOfLastWord() {
        val str = "    fly me   to   the moon  "
        var count = 0
        for (i in str.length - 1 downTo 0) {
            val c = str[i]
            if (c != ' ') {
                count++
            } else {
                if (count > 0) {
                    println("==>> Length of last word :$count")
                    return
                }
            }
        }
    }
}